hashes/src/util.rs

```diff
@@ -110,28 +110,77 @@ macro_rules! engine_input_impl(
 
 
 /// Creates a new newtype around a [`Hash`] type.
```

Comment:
perhaps
```
/// Creates a new `$newtype`, which is a wrapper around a [`Hash`] type (`$hash`).
```

Reply:
On `hash_newtype_get_direction`, also a possible alternate name for this macro is `hash_newtype_forbid_duplicate_direction_attributes`

Reply:
I think after refactoring this should be sufficiently obvious?